contrib/go-chi/chi/chitrace.go

```diff
@@ -0,0 +1,67 @@
+// Package chi provides tracing middleware for the Chi web framework.
+package chi
+
+import (
+	"context"
+	"net/http"
+	"strconv"
+	"strings"
+
+	ddtrace "github.com/DataDog/dd-trace-go/opentracing"
+	"github.com/DataDog/dd-trace-go/tracer/ext"
+	"github.com/go-chi/chi"
+	"github.com/go-chi/chi/middleware"
+	opentracing "github.com/opentracing/opentracing-go"
+	otext "github.com/opentracing/opentracing-go/ext"
+)
+
+func Middleware(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
```

Comment:
I think it might be possible to get rid of 70% of this code by using the [`internal.TraceAndServe`](https://github.com/DataDog/dd-trace-go/blob/606890436f83b6ea4421e1710a6b3629b62fee13/contrib/net/http/http.go#L42-L48) function. Checkout [net/http](https://github.com/DataDog/dd-trace-go/blob/606890436f83b6ea4421e1710a6b3629b62fee13/contrib/net/http/http.go) and [gorilla/mux](https://github.com/DataDog/dd-trace-go/blob/606890436f83b6ea4421e1710a6b3629b62fee13/contrib/gorilla/mux/mux.go) for inspiration.

Reply:
_(edited above message)_

Reply:
@gbbr Maybe I'm not reading the code correctly, but it looks like `internal.Trace` is *not* using opentracing.

Reply:
You're right, and that's fine... We will be refactoring the internals soon, so it's safe to use that.